pkg/golinters/errchkjson.go

```diff
@@ -0,0 +1,80 @@
+package golinters
+
+import (
+	"github.com/breml/errchkjson"
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
+)
+
+func NewErrChkJSONFuncName(linters *config.Linters,
+	cfg *config.ErrChkJSONSettings,
+	errcheckCfg *config.ErrcheckSettings,
+) *goanalysis.Linter {
+	a := errchkjson.NewAnalyzer()
+
+	var omitSafe bool
+	var reportNoExported bool
+	if cfg != nil {
+		omitSafe = cfg.OmitSafe
+		reportNoExported = cfg.ReportNoExported
+	}
+
+	// Modify errcheck config if this linter is enabled and OmitSafe is false
+	if isEnabled(linters, a.Name) && !omitSafe {
+		if errcheckCfg == nil {
+			errcheckCfg = &config.ErrcheckSettings{}
+		}
+		errcheckCfg.ExcludeFunctions = append(errcheckCfg.ExcludeFunctions,
+			"encoding/json.Marshal",
+			"encoding/json.MarshalIndent",
+			"(*encoding/json.Encoder).Encode",
+		)
+	}
+
```

Comment:
As I already said, a linter cannot interact with another linter.

The fact to create links between 2 linters means that 1 linter can be a part of the other linter.

Reply:
@ldez Thanks for having a look at my PR again, I really appreciate it.

> As I already said, a linter cannot interact with another linter.

I do not exactly understand what you mean by "cannot". My understanding of "cannot" is, that it is technically not possible. But I don't think this is true, because the implementation of the PR does work as intended.
Or do you mean "cannot" in the sense of "MUST NOT" (or "SHALL NOT") as defined in https://datatracker.ietf.org/doc/html/rfc2119? That is: it is prohibited by guideline (even though, it could technically be possible).
If it is the later, I would like to know, if this guideline is written down somewhere and I would like to better understand the reasoning behind it. I understand, that in some cases the exclude patterns have been used to handle exactly such situations, e.g. for duplicates between gosec and errcheck (see `EXC0008`).

> The fact to create links between 2 linters means that 1 linter can be a part of the other linter.
So what is your recommendation? I can see the following options (in no particular order):

1. Add the functionality of errchkjson to errcheck. I do not know, if this special handling is in the scope of errcheck, because it defines its scope pretty general with "... checking for unchecked errors ..." and as far as I can tell, there is currently no special handling based on the input arguments in errcheck. Additionally, errchkjson also performs checks on the input types for the json encoding functions (can a type be encoded) and this is cleary out of scope for errcheck and would have to stay in this linter.
2. Extend errchkjson to also include everything that is done by errcheck so the user is free to choose one linter or the other depending of the fact, if he wants to have the special handling of the json encoding functions. In my opinion, this does not make any sense.
3. We add the necessary exceptions to the ExcludePatterns in `pkg/config/issues.go`, but I am not sure how exactly this would work given the fact, that the exclude pattern should only be added under special circumstances (both linters enabled and omitSafe not true).
4. Add the errchkjson linter (without adding config settings to errcheck) and mention in the documentation / .golangci-lint.yaml, how the user can add the necessary excludes. This would work, but it is not really user friendly.
5. Keep the current implementation.